invoices/sql_migration_test.go

```diff
@@ -0,0 +1,434 @@
+package invoices
+
+import (
+	"context"
+	crand "crypto/rand"
+	"database/sql"
+	"math/rand"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/lightningnetwork/lnd/channeldb/models"
+	"github.com/lightningnetwork/lnd/clock"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/record"
+	"github.com/lightningnetwork/lnd/sqldb"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	// testHtlcIDSequence is a global counter for generating unique HTLC
+	// IDs.
+	testHtlcIDSequence uint64
+)
+
+// generateTestInvoice generates a random invoice for testing purposes. The
+// invoice will have a random state and random HTLCs. If mpp is true, the
+// invoice will have the MPP feature bit set. If amp is true, the invoice will
+// have the AMP feature bit set and contain AMP HTLCs. Note that this function
+// is not supposed to generate correct invoices, but rather random invoices for
+// testing the SQL migration.
+func generateTestInvoice(t *testing.T, mpp bool, amp bool) *Invoice {
```

Comment:
Since this is just a migration code test, and therefore its expected lifetime is small I don't think you should change this, however, it is worth noting that much of this code could have been avoided by using [rapid](https://github.com/flyingmutant/rapid) for all of the generation. I mention this just so that next time you encounter the need for this you need not roll your own random gen stuff. We introduced it in #9032